src/dbnode/network/server/tchannelthrift/convert/convert_test.go

```diff
@@ -253,6 +258,16 @@ func TestConvertAggregateRawQueryRequest(t *testing.T) {
 	}
 }
 
+func TestToRPCError(t *testing.T) {
+	limitErr := limits.NewQueryLimitExceededError("limit")
+	invalidParamsErr := xerrors.NewInvalidParamsError(errors.New("param"))
+
+	require.Equal(t, convert.ToRPCError(limitErr), tterrors.NewResourceExhaustedError(limitErr))
+	require.Equal(t, convert.ToRPCError(invalidParamsErr), tterrors.NewBadRequestError(invalidParamsErr))
+	require.Equal(t, convert.ToRPCError(stdctx.Canceled), tterrors.NewTimeoutError(stdctx.Canceled))
+	require.Equal(t, convert.ToRPCError(stdctx.DeadlineExceeded), tterrors.NewTimeoutError(stdctx.DeadlineExceeded))
```

Comment:
nit: maybe verify wrapped errors too? i know it's not super common in m3

Reply:
had to add a function to xerrors to properly loop through inner errors